experiment/api.js

```diff
@@ -42,9 +42,14 @@ class API extends ExtensionAPI { // eslint-disable-line no-unused-vars
       }
     }, manifestOptions);
 
-    if (this.sectionOptions.icon && !this.sectionOptions.icon.startsWith("moz-extension://")) {
-      this.sectionOptions.icon = this.extension.getURL(this.sectionOptions.icon);
-    }
+    this.wrapExtensionUrl(this.sectionOptions, "icon");
+    this.wrapExtensionUrl(this.sectionOptions.emptyState, "icon");
+  }
+
+  wrapExtensionUrl(object, prop) {
+    const isUri = typeof object[prop] === "string" &&
+      !object[prop].startsWith("moz-extension://");
+    object[prop] = isUri ? this.extension.getURL(object[prop]) : object[prop];
```

Comment:
nit: this looks to unnecessarily set the prop with the same value in the `false` case